examples/jsm/objects/BatchedMesh.js

```diff
@@ -0,0 +1,461 @@
+
+		this._vertexStarts.push( this._currentVertexCount );
+		this._vertexCounts.push( srcPositionAttribute.count );
+
+		if ( hasIndex ) {
+
+			this._indexStarts.push( this._currentIndexCount );
+			this._indexCounts.push( srcIndex.count );
+
+		}
+
+		this._visibilities.push( true );
+		this._alives.push( true );
+
+		// @TODO: Error handling if exceeding maxVertexCount or maxIndexCount
+
+		for ( const attributeName in geometry.attributes ) {
+
+			const srcAttribute = geometry.getAttribute( attributeName );
+			const dstAttribute = this.geometry.getAttribute( attributeName );
+
+			dstAttribute.array.set( srcAttribute.array, this._currentVertexCount * dstAttribute.itemSize );
+			dstAttribute.needsUpdate = true;
+
+		}
+
+		if ( hasIndex ) {
+
+			for ( let i = 0; i < srcIndex.count; i ++ ) {
+
+				dstIndex.setX( this._currentIndexCount + i, this._currentVertexCount + srcIndex.getX( i ) );
+
+			}
+
+			this._currentIndexCount += srcIndex.count;
+			dstIndex.needsUpdate = true;
+
+		}
+
+		const geometryId = this._currentGeometryCount;
+		this._currentGeometryCount ++;
+
+		const idAttribute = this.geometry.getAttribute( 'id' );
+
+		for ( let i = 0; i < srcPositionAttribute.count; i ++ ) {
+
+			idAttribute.setX( this._currentVertexCount + i, geometryId );
+
+		}
+
+		idAttribute.needsUpdate = true;
+
+		this._currentVertexCount += srcPositionAttribute.count;
+
+		this._matrices.push( new Matrix4() );
+		_identityMatrix.toArray( this._matricesArray, geometryId * 16 );
+		this._matricesTexture.needsUpdate = true;
+
+		return geometryId;
+
+	}
+
+	discardGeometry( geometryId ) {
+
+		if ( geometryId >= this._alives.length || this._alives[ geometryId ] === false ) {
+
+			// @TODO: Warning?
+			return this;
+
+		}
+
+		this._alives[ geometryId ] = false;
+		_zeroMatrix.toArray( this._matricesArray, geometryId * 16 );
+		this._matricesTexture.needsUpdate = true;
+
+		// User needs to call optimize() to pack the data.
+
+		return this;
+
+	}
+
+	// @TODO: Rename to better name, like deflag or pack?
+	optimize() {
+
+		// @TODO: Implement
+
+		return this;
+
+	}
+
+	setMatrixAt( geometryId, matrix ) {
+
+		// @TODO: Map geometryId to index of the arrays because
+		//        optimize() can make geometryId mismatch the index
+
+		if ( geometryId >= this._matrices.length || this._alives[ geometryId ] === false ) {
+
+			// @TODO: Warning?
+			return this;
+
+		}
+
+		this._matrices[ geometryId ].copy( matrix );
+
+		if ( this._visibilities[ geometryId ] === true ) {
+
+			matrix.toArray( this._matricesArray, geometryId * 16 );
+			this._matricesTexture.needsUpdate = true;
+
+		}
+
+		return this;
+
+	}
+
+	getMatrixAt( geometryId, matrix ) {
+
+		if ( geometryId >= this._matrices.length || this._alives[ geometryId ] === false ) {
+
+			// @TODO: Warning?
+			return matrix;
+
+		}
+
+		return matrix.copy( this._matrices[ geometryId ] );
+
+	}
+
+	setVisibilityAt( geometryId, visiblity ) {
```

Comment:
I'm undecided on this ... would `.setVisibleAt` be better? Just to match `.visible`.

Reply:
OK, updated. Thanks.